WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift

```diff
@@ -47,8 +49,26 @@ public class ProductsScreen: ScreenObject {
     }
 
     @discardableResult
-    public func selectProduct(atIndex index: Int) throws -> SingleProductScreen {
-        XCUIApplication().tables.cells.element(boundBy: index).tap()
+    public func selectProductByIndex(atIndex index: Int) throws -> SingleProductScreen {
+        app.tables.cells.element(boundBy: index).tap()
         return try SingleProductScreen()
     }
+
+    public func selectProductByName(name: String) throws -> SingleProductScreen {
+        app.tables.cells.staticTexts[name].tap()
+        return try SingleProductScreen()
+    }
+
+    public func verifyProductOnProductsScreen(count: Int, name: String, status: String) throws -> ProductsScreen {
+        XCTAssertTrue(try app.getTextVisibilityCount(text: name) == 1, "Product name does not exist!")
+        XCTAssertTrue(app.tables.cells.count == count, "Expecting \(count) products, got \(app.tables.cells.count) instead!")
+        XCTAssertTrue(try app.getTextVisibilityCount(text: status) > 0, "Stock status does not exist!")
+
+        return try ProductsScreen()
+    }
+
+    public func verifyProductScreenLoaded() throws -> ProductsScreen {
+        XCTAssertTrue(isLoaded)
+        return try ProductsScreen()
+    }
```

Comment:
This is applicable to all three `verify...` methods that were added.

By the time when these methods are called, the current screen is already instantiated. And then it's instantiated again (`return try ProductsScreen()`), which takes quite a bit of time. This change:

```suggestion
    public func verifyProductScreenLoaded() throws -> Self {
        XCTAssertTrue(isLoaded)
        return self
    }
```

Shaves of 25 seconds on iPhone 13 (101 sec ~> 75 sec).